crates/astria-conductor/src/executor.rs

```diff
@@ -272,50 +311,26 @@ impl<C: ExecutionClientV1Alpha1> Executor<C> {
                 // try executing the block as it hasn't been executed before
                 // execute_block will check if our namespace has txs; if so, it'll return the
                 // resulting execution block hash, otherwise None
-                let Some(execution_block_hash) = self
-                    .execute_block(block)
+                let Some(executed_block) = self
+                    .execute_block(block.clone())
                     .await
                     .wrap_err("failed to execute block")?
                 else {
                     // no txs for our namespace, nothing to do
-                    debug!("execute_block returned None; skipping finalize_block");
+                    debug!("execute_block returned None; skipping call_update_commitment_state");
                     return Ok(());
                 };
-
-                // finalize the block after it's been executed
-                self.finalize_block(execution_block_hash, sequencer_block_hash)
-                    .await?;
+                self.update_commitment_state(
+                    executed_block,
+                    self.commitment_state.soft.clone().unwrap(),
```

Comment:
I think we have the actions here flipped, if we just executed update both, otherwise only firm.